compiler/rustc_data_structures/src/sync.rs

```diff
@@ -26,6 +26,30 @@ use std::panic::{catch_unwind, resume_unwind, AssertUnwindSafe};
 pub use std::sync::atomic::Ordering;
 pub use std::sync::atomic::Ordering::SeqCst;
 
+// 0: uninitialized
+// 1: single-thread mode
+// 2: multiple-thread mode
+static PARALLEL: std::sync::atomic::AtomicUsize = std::sync::atomic::AtomicUsize::new(0);
```

Comment:
This controls synchronization not parallelism. I'd put it in a `mode` module with `get` and `set` functions.

Reply:
Done